ci/process_notebooks.py

```diff
@@ -158,34 +159,47 @@ def extract_solutions(nb, nb_dir, nb_name):
             cell_id = hashlib.sha1(cell_source.encode("utf-8")).hexdigest()[:8]
 
             # Extract image data from the cell outputs
-            cell_images = []
+            cell_images = {}
             for j, output in enumerate(cell.get("outputs", [])):
 
                 fname = f"static/{nb_name}_Solution_{cell_id}_{j}.png"
                 try:
                     image_data = a2b_base64(output["data"]["image/png"])
                 except KeyError:
                     continue
-                solution_resources[fname] = image_data
-                cell_images.append(fname)
+                cell_images[fname] = Image.open(BytesIO(image_data))
+                static_images.update(cell_images)
```

Comment:
do you need to do this inside the for loop?

Reply:
Nope.